docs/docs/guides/integrate/identity-providers/gitlab.md

```diff
@@ -0,0 +1,103 @@
+
+## GitLab Configuration
+
+### Register a new application
+
+1. Login to [gitlab.com](https://gitlab.com)
+2. Select [Edit Profile](https://gitlab.com/-/profile)
+3. Click on [Applications](https://gitlab.com/-/profile/applications) in the side navigation
+
+For **GitLab Self-Hosted** go to your GitLab self-hosted instance and follow the same steps as for GitLab.
+
+Fill in the application name.
+
+You have to add the redirect URI, where GitLab should redirect, after the user has authenticated himself.
+In this example our test instance has the domain `https://acme-gzoe4x.zitadel.cloud`.
+This results in the following redirect URI:
+ `https://acme-gzoe4x.zitadel.cloud/ui/login/login/externalidp/callback`
+
+:::info
+To adapt this for you setup just replace the domain
+:::
+
+![Register an OAuth application](/img/guides/gitlab_app_registration.png)
+
+### Client ID and Secret
+
+After clicking "Save application", you will see the detail page of the application you have just created.
+To be able to connect GitLab to ZITADEL you will need a client ID and a client secret. 
+Save the ID and the Secret, you will not be able to copy the secret again, if you lose it you have to generate a new one.
+
+![Client ID and Secret](/img/guides/gitlab_app_id_secret.png)
+
+## ZITADEL Configuration
+
+### Create new GitHub Provider
+
+Go to the settings of your ZITADEL instance or the organization where you like to add a new GitHub provider.
+Choose the GitHub provider template. This template has everything you need preconfigured. 
+Add the client ID and secret you have created in the GitHub Application.
+
+You can configure the following settings if you like, a useful default will be filled if you don't change anything:
+
+**Scopes**: The scopes define which scopes will be sent to the provider, `openid`, `profile`, and `email` are prefilled. This informations will be taken to create/update the user within ZITADEL.
+
+<GeneralConfigDescription name="GeneralConfigDescription" />
+
+![GitHub Provider](/img/guides/zitadel_gitlab_create_provider.png)
+
+### Activate IdP
+
+Once you created the IdP you need to activate it, to make it usable for your users.
+
+![Activate the GitHub](/img/guides/zitadel_activate_gitlab.png)
+
+## Test the setup
+
+To test the setup use incognito mode and browse to your login page.
+If you succeeded you should see a new button which should redirect you to your GitHub.
+
+Per default the login of your instance will be shown, read the following section on how to trigger it for a specific organization: [Organization Scope](./general#trigger-configuration-on-the-login-for-a-specific-organization)
+
+![GitHub Button](/img/guides/zitadel_login_gitlab.png)
+
+![GitHub Login](/img/guides/gitlab_login.png)
+
+If the user is not yet linked in ZITADEL the user will see the screen below.
+Because GitHub is an OAuth provider and oAuth does not provide a standardized way to get the user data, not all the data can be taken over. 
```

Comment:
See suggestion on GitHub.
Replace GitHub here.
Maybe centralize?

Reply:
comment is obsolete if we implement https://github.com/zitadel/zitadel/pull/5565#pullrequestreview-1374481217